master/internal/rm/kubernetesrm/pod.go

```diff
@@ -317,55 +238,125 @@ func (p *pod) receivePodStatusUpdate(ctx *actor.Context, msg podStatusUpdate) er
 			// determined containers generates an exit code. To check if this is
 			// the case we check if a deletion timestamp has been set.
 			if p.pod.ObjectMeta.DeletionTimestamp != nil {
-				ctx.Log().Info("unable to get exit code for pod, setting exit code to 1025")
+				p.syslog.Info("unable to get exit code for pod, setting exit code to 1025")
 				exitCode = 1025
 				exitMessage = "unable to get exit code or exit message from pod"
 			} else {
-				return err
+				return p.container.State, err
 			}
 		}
 
-		ctx.Log().Infof("transitioning pod state from %s to %s", p.container.State, containerState)
+		p.syslog.Infof("transitioning pod state from %s to %s", p.container.State, containerState)
 		p.container = p.container.Transition(cproto.Terminated)
 
 		var resourcesStopped sproto.ResourcesStopped
 		switch exitCode {
 		case aproto.SuccessExitCode:
-			ctx.Log().Infof("pod exited successfully")
+			p.syslog.Infof("pod exited successfully")
 		default:
-			ctx.Log().Infof("pod failed with exit code: %d %s", exitCode, exitMessage)
+			p.syslog.Infof("pod failed with exit code: %d %s", exitCode, exitMessage)
 			resourcesStopped.Failure = sproto.NewResourcesFailure(
 				sproto.ResourcesFailed,
 				exitMessage,
 				ptrs.Ptr(sproto.ExitCode(exitCode)))
 		}
-		p.informTaskResourcesStopped(ctx, resourcesStopped)
-		ctx.Self().Stop()
+		p.informTaskResourcesStopped(resourcesStopped)
+		return p.container.State, nil
 
 	default:
 		panic(fmt.Sprintf("unexpected container state %s", containerState))
 	}
 
-	return nil
+	return p.container.State, nil
 }
 
-func (p *pod) deleteKubernetesResources(ctx *actor.Context) {
+// TODO(!!!): consistency around locking / non-locking and caps.
```

Comment:
If this is meant to live beyond this project, is it worth being a bit more descriptive?

Reply:
ah, i just meant to fix this, oops.

Reply:
Right on. There was [another TODO](https://github.com/determined-ai/determined/pull/7617/files#diff-fd108f0536582e46db7069c4ed42c1390e27b7d835c4a7be71de6e42e909d4f4R469), too. Also meant to be fixed?